Added Assigment-3

akashdoppalapudi opened the pull request:
Created Design Diagrams in MySQL workbench and added them.
Generated SQL files for the schema and added them as well.

akashdoppalapudi committed:
added pdf file

akashdoppalapudi committed:
added schema diagrams and sql files

husunu-technovert commented on assignment-03/technovert-cup-cricket-tournament-schema.sql:
```diff
@@ -0,0 +1,139 @@
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `match` (
+  `MatchID` tinyint unsigned NOT NULL AUTO_INCREMENT,
```
1. Why tinyint for pk column? In future if we want to store yearly data, tinyint will not be enough
2. This table seems incomplete. Generally a match will be between two teams right. I don't see that here.
3. where can i see playing 11 for each match?

akashdoppalapudi replied:
1. Used INT instead of TINYINT
2. added columns for team1 and team2
3. using the team id we can see the players in the team.

husunu-technovert commented on assignment-03/technovert-cup-cricket-tournament-schema.sql:
```diff
@@ -0,0 +1,139 @@
+  `PlayerID` tinyint unsigned NOT NULL AUTO_INCREMENT,
+  `FirstName` varchar(50) NOT NULL,
+  `LastName` varchar(50) NOT NULL,
+  `BirthDate` date NOT NULL,
```
1. Rename BirthDate to DateOfBirth
2. Give better name for Position. (eg: Role)
3. Follow proper naming convention for constraints. (fk _ {fkTable} _ {currentTable} _ {columnName})
4. Change PK column to INT instead of tinyint

akashdoppalapudi replied:
1. Renamed BirthDate to DateOfBirth
2. Changed Position to Role
3. flowed proper naming convention for constraints
4. changed PK to INT

husunu-technovert commented on assignment-03/technovert-cup-cricket-tournament-schema.sql:
```diff
@@ -0,0 +1,139 @@
+  `MatchID` tinyint unsigned NOT NULL,
+  `PlayerID` tinyint unsigned NOT NULL,
+  `Score` int unsigned NOT NULL,
+  `Status` enum('Not Out','Run Out','Caught','Bowled','Stump Out') NOT NULL DEFAULT 'Not Out',
```
1. MatchID > tinyint is not appropriate
2. Why this table doesn't have pk?
3. Status - enum is not appropriate, we can store in another table
4. Out By - what if he is stumpped - where are you storing stumpped by
5. What if there are two people involved in runouts?
6. Where are you storing wickets taken (or) runs conceded
7. Where can i see who took catch?

akashdoppalapudi replied:
1. Used INT for PK and other FK
2. introduced a PK
3. used another table to store batter status
4. OutBy is used to store details like who stumped, caught or made the runout
5.  Bowler represents the player who bowled when the player got out.
6. since they are nullable.. If player is not out bowler and out by will be kept null
7. we can count wickets taken by a player by number of times a bowler appeared in scores table in a particular match.

husunu-technovert commented on assignment-03/technovert-cup-cricket-tournament-schema.sql:
```diff
@@ -0,0 +1,139 @@
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `team` (
+  `TeamID` tinyint unsigned NOT NULL AUTO_INCREMENT,
```
tinyint is not appropriate

akashdoppalapudi replied:
used INT

akashdoppalapudi committed:
changed TINYINT to INT in cricket database

also added team1 and team2 ids in match table

akashdoppalapudi committed:
Updated player table and others

1. Renamed BirthDate to DateOfBirth
2. Changed Position to Role
3. flowed proper naming convention for constraints
4. changed PK to INT

akashdoppalapudi committed:
created new table to describe batter status

akashdoppalapudi committed:
made outby and bowler columns nullable
